papas/simulator.py

```diff
@@ -208,61 +227,70 @@ def simulate_hadron(self, ptc):
                     smeared = self.smear_cluster(cluster, hcal, acceptance=ecal)
                     if smeared:
                         ptc.clusters_smeared[smeared.layer] = smeared
+
         cluster = self.make_cluster(ptc, 'hcal', 1-frac_ecal)
         smeared = self.smear_cluster(cluster, hcal)
         if smeared:
             ptc.clusters_smeared[smeared.layer] = smeared
-        if ptc.q()!=0:
-            smeared_track = self.smear_track(ptc.track,
-                                             self.detector.elements['tracker'])
-            if smeared_track:
-                ptc.track_smeared = smeared_track
 
     def simulate_muon(self, ptc):
+        pdebugger.info("Simulating Muon")
         self.propagate(ptc)
         smeared_track = self.smear_track(ptc.track,
                                          self.detector.elements['tracker'])
         if smeared_track:
             ptc.track_smeared = smeared_track
 
     def smear_muon(self, ptc):
+        pdebugger.info("Smearing Muon");
         self.propagate(ptc)
+        if ptc.q()!=0:
+            pdebugger.info("Made " + ptc.track.__str__())
         smeared = copy.deepcopy(ptc)
         return smeared
 
     def smear_electron(self, ptc):
+        pdebugger.info("Smearing Electron");
         ecal = self.detector.elements['ecal']
         self.prop_helix.propagate_one(ptc,
                                       ecal.volume.inner,
                                       self.detector.elements['field'].magnitude )
+        if ptc.q()!=0:
+            pdebugger.info("Made " + ptc.track.__str__())
         smeared = copy.deepcopy(ptc)
         return smeared
     
     def propagate_muon(self, ptc):
+        pdebugger.info("Propogate Muon")
         self.propagate(ptc)
         return 
     
     def propagate_electron(self, ptc):
+        pdebugger.info("Propogate Electron")
         ecal = self.detector.elements['ecal']
         self.prop_helix.propagate_one(ptc,
                                       ecal.volume.inner,
                                       self.detector.elements['field'].magnitude )
         return    
     
-    def simulate(self, ptcs, do_reconstruct = True):
+    def simulate(self, ptcs):
         self.reset()
         self.ptcs = []
         smeared = []
+
+        #newsort
+        for gen_ptc in sorted(ptcs, key = lambda ptc: ptc.uniqueid):
+            pdebugger.info(str('{}'.format(gen_ptc)))        
         for gen_ptc in ptcs:
             ptc = pfsimparticle(gen_ptc)
             if ptc.pdgid() == 22:
                 self.simulate_photon(ptc)
-            elif abs(ptc.pdgid()) == 11:
+            elif abs(ptc.pdgid()) == 11: #check with colin
```

Comment:
My last code contained smear_electron(ptc) instead of propagate electron. Similarly for Muons. Was wanting to understand why the change and what implications (if any) it has.